client/components/utils/add-comment-plugin.js

```diff
@@ -79,21 +79,17 @@ export default function addCommentPlugin(vueInstance) {
   });
 }
 
-export function addHighlight(keys, schema, state, from, to, dispatch) {
-  // "tr" is a ProseMirror transaction.
-  const {tr} = state;
+export function addHighlight(keys, schema, tr, from, to, dispatch) {
   const attrs = {"highlight-keys": keys};
   tr.setMeta("addToHistory", false);
-  return dispatch(tr.addMark(from, to, schema.marks.highlight.create(attrs)));
+  return tr.addHighlightMark(from, to, schema.marks.highlight.create(attrs));
```

Comment:
Why don't you have to call `dispatch` anymore?

Reply:
Answer: because we are dispatching later on multiple steps at once, no need to do it one by one.